train.py

```diff
@@ -140,7 +140,7 @@ def import_model_class_from_model_name_or_path(
 
         return UMT5EncoderModel
     elif model_class == "ChatGLMModel":
-        from diffusers import ChatGLMModel
+         from diffusers.pipelines.kolors.text_encoder import ChatGLMModel
```

Comment:
i'm not sure this is needed - git master diffusers does not error. can you check this?

Reply:
![image](https://github.com/user-attachments/assets/90aced9a-6d91-4bf6-a8fc-a5b52771be65)
![image](https://github.com/user-attachments/assets/502c5ba2-1596-40fa-89fe-66b66871cb12)
my diffusers version is 0.30.0.dev0  and  I use the python command like the above image,you can see the `from diffusers import ChatGLMModel`  report `ImportError: cannot import name 'ChatGLMModel' from 'diffusers'`

Reply:
![image](https://github.com/user-attachments/assets/522a7e8c-dc28-4dc8-9796-88cc3e77c116)
`from diffusers.pipelines.kolors.text_encoder import ChatGLMModel` this code can import the ChatGLMModel;

Reply:
```
Disabled Xformers on MacOS, as it is not yet supported.
Overridden MIXED_PRECISION to 'no' for MacOS, as autocast is not supported by MPS.
2024-08-06 20:36:19,609 [WARNING] (ArgsParser) MPS may benefit from the use of --unet_attention_slice for memory savings at the cost of speed.
2024-08-06 20:36:19,609 [INFO] (ArgsParser) VAE Model: madebyollin/sdxl-vae-fp16-fix
2024-08-06 20:36:19,609 [INFO] (ArgsParser) Default VAE Cache location: 
2024-08-06 20:36:19,609 [INFO] (ArgsParser) Text Cache location: cache
2024-08-06 20:36:19,615 [INFO] (__main__) Load tokenizers
2024-08-06 20:36:20,062 [INFO] (__main__) Loading Kolors ChatGLM language model..
2024-08-06 20:36:20,383 [INFO] (__main__) Loading ChatGLM language model from kwai-kolors/kolors-diffusers/text_encoder..
Downloading shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 16384.00it/s]
Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:04<00:00,  1.58s/it]
2024-08-06 20:36:25,745 [INFO] (__main__) Load VAE: madebyollin/sdxl-vae-fp16-fix
2024-08-06 20:36:25,835 [WARNING] (__main__) Couldn't load VAE with default path. Trying without a subfolder..
2024-08-06 20:36:26,080 [INFO] (__main__) Moving text encoder to GPU.
2024-08-06 20:36:28,426 [INFO] (__main__) Loading VAE onto accelerator, converting from torch.float32 to torch.bfloat16
```

with current main branch, it works for me. i'm not sure why yours isn't, but perhaps you're not on the actual latest main git diffusers?

Reply:
```
% python
Python 3.10.14 (main, Mar 19 2024, 21:46:16) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from diffusers import ChatGLMModel
>>> 
```

Reply:
```
diffusers                    0.30.0.dev0
```

but the way pip works you have to `pip uninstall diffusers` before trying to update to the latest git main branch. please try that first.

Reply:
![image](https://github.com/user-attachments/assets/b5440513-48ff-499b-b310-a898bb2305bc)
![image](https://github.com/user-attachments/assets/30d5d5a9-410d-4ab9-951e-5c7efe6f9617)
i try to` pip uninstall diffusers` and git clone the latest diffusers and cd  folder to use `pip install .` this command also report `ImportError: cannot import name 'ChatGLMModel' from 'diffusers'`

Reply:
@sayakpaul do you have any idea about this mystery?

Reply:
Cc: @asomoza

Reply:
i'll merge it but i really don't understand why it doesn't work, which in a way, bothers me :D

Reply:
After this [PR](https://github.com/huggingface/diffusers/pull/9065) was merged, we can't import  `ChatGLMModel` directly from `diffusers` because that would require to add `sentencepiece` as an additional dependency.